slave-new/process-o365/bin/process-o365.sh

```diff
@@ -0,0 +1,262 @@
+#!/bin/bash
+
+PROTOCOL_VERSION='3.0.0'
+
+function process {
+	ACCESS_TOKEN=`cat /tmp/o365-access-token`
+
+	LICENSES="licenses.csv"
+	FILE_PHOTOS="photos.csv"
+	FILE_USERS="users.json"
+
+	E_USERS_SYNC=(50 'Error during users synchronization')
+
+	HOME_DIR=`eval echo ~`
+
+	create_lock
+
+	# GET licenses:
+	curl --header "Authorization: $ACCESS_TOKEN" --header "Accept: application/json" 'https://graph.microsoft.com/v1.0/subscribedSkus' | perl -nle 'print "$1,$2" while(/"skuId"\s*:\s*"([^"]*)","skuPartNumber"\s*:\s*"([^"]*)"/g)' >> $HOME_DIR/$LICENSES
+
+	PERL_SYNCHRONIZE_USERS_SCRIPT='
+		use LWP::UserAgent;
+		use MIME::Base64;
+		use JSON;
+
+		my $licenses = {};
+		my $photos = {};
+
+		# o365
+		my @listOfUsersActual;
+		my $o365UserMapper = {};
+		my $filenameLicenses = '${HOME_DIR}/$LICENSES';
```

Comment:
If you need temporary file, rather use $WORK_DIR. It's created only for one run of slave script, therefore you are sure what files are in it and the wole dir si automatically deleted after slave script ends.